src/D2L.Security.OAuth2/OAuth2Factory.cs

```diff
@@ -0,0 +1,38 @@
+﻿using D2L.Security.OAuth2.Keys;
+using D2L.Security.OAuth2.Keys.Default;
+using D2L.Security.OAuth2.Utilities;
+
+namespace D2L.Security.OAuth2 {
+	public static class OAuth2Factory {
+		/// <summary>
+		/// Create an IKeyManagmentService and ITokenSigner given key storage
+		/// and configuration.
+		/// </summary>
+		public static void Create(
+			IPublicKeyDataProvider publicKeyDataProvider,
+			IPrivateKeyDataProvider privateKeyDataProvider,
+
+			out IKeyManagementService keyManagementService,
+			out ITokenSigner tokenSigner,
+
+			OAuth2Configuration config = null

```

Comment:
Key types would go in `OAuth2Configuration`?

Reply:
That makes sense to me.